src/com/google/enterprise/adaptor/DocumentContentTransform.java

```diff
@@ -0,0 +1,68 @@
+package com.google.enterprise.adaptor;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The document content transform can modify the content of a document.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class DocumentContentTransform extends OutputStream {
+
+  protected final Map<String, String> config;
+  protected final Metadata metadata;
+  protected final String contentType;
+  private OutputStream originalStream;
+
+  /**
+   * Constructs a document content transformer.
+   *
+   * @param config         the configuration for this instance
+   *                       (never null, but maybe empty)
+   * @param originalStream the original content stream
+   * @param contentType    the content type of the stream
+   *                       (could possibly be null or empty)
+   * @param metadata       the metadata already collected for this document
+   *                       (never null, but maybe empty)
+   */
+  public DocumentContentTransform(final Map<String, String> config,
+                                  final OutputStream originalStream, final String contentType,
+                                  final Metadata metadata) {
```

Comment:
Metadata instances are mutable.

I'm thinking whether programmers of DocumentContentTransform classes will expect to be able to change it.

At the point where a DocumentContentTransform is instantiated the HTTP headers have already been sent out, so any changes won't affect the metadata GSA receives.  However it could be used as a communication channel between DocumentContentTransform instances in a pipeline.  I wonder if that is a good thing, or if that is something that should be made more explicit.